app/jobs/batch_autoscaling.py

```diff
@@ -0,0 +1,77 @@
+import boto3
+
+def get_compute_environment_capacity(compute_environment_name, region):
+    batch_client = boto3.client('batch', region_name=region)
+    response = batch_client.describe_compute_environments(computeEnvironments=[compute_environment_name])
+    compute_resources = response['computeEnvironments'][0]['computeResources']
+    return {'minvCpus': compute_resources['minvCpus'], 'maxvCpus': compute_resources['maxvCpus']}
+
+
+def set_compute_environment_min_capacity(vcpu_min_capacity, compute_environment_name, region):
+    batch_client = boto3.client('batch', region_name=region)
+    batch_client.update_compute_environment(computeEnvironment=compute_environment_name, computeResources={'minvCpus': vcpu_min_capacity})
+
+
+def check_pending_jobs_counts(queue_name, region, job_status_list=('RUNNABLE', 'STARTING', 'RUNNING')):
+    return {job_status: get_jobs_count(queue_name, region, job_status) for job_status in job_status_list}
+
+
+def autoscale_compute_environments(batch_configurations, dry_run=False):
+    results = []
+    for bc in batch_configurations:
+        try:
+            process_results = process_batch_configuration(bc, dry_run)
+            results.append({'batch_configuration': bc, 'process_results': process_results})
+        except Exception as e:
+            error_type = str(type(e))
```

Comment:
Why is it necessary to suppress exceptions?

If the exception has to be suppressed, could you `traceback.print_exc()` or log `traceback.format_exc()`?   Having the stack trace reduces the time to identify the true cause of some bugs.

Reply:
I just want to reduce the noise in case that error `Cannot update, compute environment ... is being modified.` happens too often. I hope that's not the case, since autoscaler reevaluates every 60 seconds and it is probably time enough to make compute environment get back into a ready state. 
However I agree with you, it might be helpful to have the stack trace if something else goes wrong. I'm be adding a traceback to this results object.